### DISCUSSION TITLE
JWT 토큰을 파싱할 때, 어떤 예외가 발생할 수 있을까?

### DISCUSSION THREAD

**Discussion by beatmeJY** [dailyge/dailyge-server#35] [Information]
## 🔗 연관 PR
- [[DLG-48] 만료된 AccessToken는 RefreshToken으로 토큰을 재발급한다.](https://github.com/dailyge/dailyge-server/pull/33)

<br/><br/>

## ❓ 작성 이유
JWT를 통한 Auth 구현을 하며 JWT 토큰 파싱 도중 발생할 수 있는 예외 종류에 대해 학습하고 정리해두기 위해 작성하였습니다.

<br/><br/>

#### `IllegalArgumentException`
  - JWT이 비어있을 경우 발생합니다.
#### `ClaimJwtException`
  - JWT 권한claim 검사가 실패했을 때 발생합니다.
#### `ExpiredJwtException`
  - 유효 기간이 지난 JWT를 수신한 경우 발생합니다.
  - 해당 예외를 통해 토큰이 만료됨을 판단하여 재갱신을 수행할 수 있습니다.
#### `MalformedJwtException`
  - 토큰의 형식 등이 올바르지 않고 구조적인 문제가 있는 JWT인 경우에 발생합니다.
#### `PrematureJwtException`
  - 접근이 허용되기 전인 JWT가 수신된 경우에 발생합니다.
  - 예를 들어 특정 보안 요구사항 때문에 토큰이 발급된 직후 바로 사용되지 않도록 일정 시간의 유예 기간을 설정할 수 있습니다
  - 이 경우 서버간의 시간이 여러가지 요인으로 다를 수 있는데 토큰의 허용 시간이 현재 보다 미래일 경우 발생할 수 있습니다.
#### `SignatureException`
  - 토큰의 서명 검증이 실패한 경우에 발생합니다.
#### `UnsupportedJwtException`
  - 수신한 JWT의 형식이 애플리케이션에서 원하는 형식과 맞지 않는 경우에 발생합니다.
  - 예를 들어, 암호화된 JWT를 사용하는 애플리케이션에 암호화되지 않은 JWT가 전달되는 경우에 이 예외가 발생합니다.

<br /><br />

## 📚 참고자료

- [JWT 자바 가이드](https://medium.com/@OutOfBedlam/jwt-%EC%9E%90%EB%B0%94-%EA%B0%80%EC%9D%B4%EB%93%9C-53ccd7b2ba10)
- [JJWT GitHub](https://github.com/jwtk/jjwt?tab=readme-ov-file#quickstart)

**Comment by devjun10** [comment]
깔~끔하네요. 좋은 정보 잘 봤습니다. 👍
